src/sentry/tasks/data_export.py

```diff
@@ -1,8 +1,148 @@
+            file_name = process_issue_by_tag(data_export, tf)
+        # Create a new File object and attach it to the ExportedData
+        tf.seek(0)
+        with transaction.atomic():
+            file = File.objects.create(name=file_name, type="export.csv")
+            file.putfile(tf)
+            data_export.update(file=file)
+
+
+def process_discover_v2(data_export, file):
+    # TODO(Leander): Implement processing for Discover V2
+    return
+
+
+def process_billing_report(data_export, file):
+    # TODO(Leander): Implement processing for Billing Reports
+    return
+
+
+def process_issue_by_tag(data_export, file):
+    """
+    Convert tag payload to a CSV, returns (file_path, file_name) as a tuple
+    (Adapted from 'src/sentry/web/frontend/group_tag_export.py')
+    """
+    # Get the pertaining project
+    payload = data_export.query_info
+    project = Project.objects.get(slug=payload["project_slug"])
+
+    # Get the pertaining issue
+    group, _ = get_group_with_redirect(
+        payload["group_id"], queryset=Group.objects.filter(project=project)
+    )
+
+    # Get the pertaining key
+    key = payload["key"]
+    lookup_key = u"sentry:{0}".format(key) if tagstore.is_reserved_key(key) else key
+
+    # If the key is the 'user' tag, attach the event user
+    def attach_eventuser(items):
+        users = EventUser.for_tags(group.project_id, [i.value for i in items])
+        for item in items:
+            item._eventuser = users.get(item.value)
+
+    # Create the fields/callback lists
+    if key == "user":
+        callbacks = [attach_eventuser]
+        fields = [
+            "value",
+            "id",
+            "email",
+            "username",
+            "ip_address",
+            "times_seen",
+            "last_seen",
+            "first_seen",
+        ]
+    else:
+        callbacks = []
+        fields = ["value", "times_seen", "last_seen", "first_seen"]
+
+    # Example file name: ISSUE_BY_TAG-project10-user__721.csv
+    file_details = u"{}-{}__{}".format(payload["project_slug"], key, data_export.id)
+    file_name = get_file_name(ExportQueryType.ISSUE_BY_TAG_STR, file_details)
+
+    # Iterate through all the GroupTagValues
+    with start_writer(file, fields) as writer:
+        iteration = 0
+        while True:
+            gtv_list = tagstore.get_group_tag_value_iter(
+                project_id=group.project_id,
+                group_id=group.id,
+                environment_id=None,
+                key=lookup_key,
+                callbacks=callbacks,
+                offset=SNUBA_MAX_RESULTS * iteration,
+            )
+            gtv_list_raw = [serialize_issue_by_tag(key, item) for item in gtv_list]
+            if len(gtv_list_raw) == 0:
+                break
+            writer.writerows(gtv_list_raw)
+            iteration += 1
+    return file_name
+
+
+@contextmanager
+def start_writer(file, fields):
+    writer = csv.DictWriter(file, fields)
+    writer.writeheader()
+    yield writer
```

Comment:
@wedamija I'm trying to abstract as much as possible to be used in the other `processing` functions (e.g. Billing, Discover v2), let me know if this is a good way to do that.